### PR TITLE
feat: add custom colors

### PR DESCRIPTION
**Pull request by comfysage**
- **refactor: separate msg and ascii cat**
- **feat: add custom colors flag**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -126,5 +124,27 @@ fn main() -> Result<()> {
 		chars.arrow,
 	);
 
+	let mut color1 = console::Color::White;
+	let mut color2 = console::Color::White;
+	if let Some(colors) = args.colors {
+		if colors.len() > 1 {
+			color1 = console::Color::Color256(colors[0]);
+			color2 = console::Color::Color256(colors[1]);
+		}
```
Comment (uncenter) [inline src/main.rs:133]: Doesn't clap already enforce this? Why do we need to check here - or is `num_args` the *max* args?

File: src/main.rs [hunk 2]
```diff
@@ -61,6 +61,9 @@ struct Cli {
 	/// Enable kittythink mode (thought bubble)
 	#[clap(long, short)]
 	think: bool,
+	/// use custom colors
+	#[arg(long, short, num_args(2))]
+	colors: Option<Vec<u8>>,
```
Comment (uncenter) [inline src/main.rs:66]: The issue with this option is it is pretty vague how to use it without knowing the underlying code. Maybe a longer docstring could help?
```suggestion
	/// Use custom colors. The first colors the message, the second colors the cat
	#[arg(long, short, num_args(2))]
	colors: Option<Vec<u8>>,
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Comment by uncenter** [comment]
Do you mind adding a section to the README on this flag?

**Comment by comfysage** [comment]
> Do you mind adding a section to the README on this flag?

sure

**Review by uncenter — APPROVED** [review]

**Comment by uncenter** [comment]
Thank you!

**Pull request merged**